### PR TITLE
Resolve 30: reproduce simulation from first paper

### LINKED ISSUE #30: Reproduce simulation from first paper on current main branch
features used in the first paper have been implemented on the master branch (issue #364 in old GitLab.)
It should be possible to reproduce the simulation and results from the branch that was used for the paper on the master branch.

### PR DESCRIPTION
**Pull request by dabele**
## Changes

sim application with basic command line interface that reproduces the results of our first published paper. Minor differences between the two versions:

- fixed a bug in the range of ages that commute, on the paper branch the oldest (partial) commuter age group was unintentionally excluded.
- read_population seems to produce slightly different initial values, maybe due to interpolation, seems to affect mainly the number of recovered. is this expected, @WadimKoslow?
- holidays on master branch do not include weekends before and after. holidays on paper branch include both weekends before and after.
- when dynamic NPIs are applied, redundant dampings are removed (i.e. two dampings of same type, level, value). This includes not just dampings that were added by dynamic NPIs but also static dampings that were set by the user before starting the simulation, e.g. two of our seasonal scenarios if the values are identical. unless I'm missing something, this has no effect on the simulation result but may be unexpected when inspecting the dampings after the simulation.

## Merge Request - GuideLine Checklist 

**Guideline** to check code before resolve WIP and approval, respectively.
As many checkboxes as possible should be ticked.

### Checks by code author:
* [x] There is at least one issue associated with the pull request.
* [x] The branch follows the naming conventions as defined in the [git workflow](git-workflow).
* [x] New code adheres with the [coding guidelines](coding-guidelines)
* [ ] Tests for new functionality has been added
* [x] A local test was succesful
* [x] There is appropriate **documentation** of your work. (use doxygen style comments)
* [ ] If new third party software is used, did you pay attention to its license? Please remember to add it to the wiki after successful merging.
* [ ] If new mathematical methods or epidemiological terms are used, has the glossary been updated ? Did you provide further documentation ?
 is present or referenced. Please provide your references.
* [ ] The following questions are addressed in the documentation*:  Developers (what did you do?, how can it be maintained?), For users (how to use your work?), For admins (how to install and configure your work?)
* [x] For documentation: Please write or update the Readme in the current working directory!

### Checks by code reviewer(s):
* [x] Is the code clean of development artifacts e.g., unnecessary comments, prints, ...
* [x] The ticket goals for each associated issue are reached or problems are clearly addressed (i.e., a new issue was introduced).
* [ ] There are appropriate **unit tests** and they pass.
* [x] The git history is clean and linearized for the merge request.
* [x] Coverage report for new code is acceptable. 

closes #30

### CODE REVIEW

File: cpp/sims/paper_202011.cpp [hunk 1]
```diff
@@ -0,0 +1,877 @@
+/**
+* Simulation application that was used to produce results of the following publication(s):
```
Comment (mknaranja) [inline cpp/sims/paper_202011.cpp:2]:
```suggestion
* Reworked version of simulation application that was used to produce results of the following publication(s):
```

**Comment by WadimKoslow** [comment]
what do you mean by different initial values? Different for each run, or different compared to the results of our last paper?

**Comment by dabele** [comment]
> what do you mean by different initial values? Different for each run, or different compared to the results of our last paper?

calling read_population on current main gives a different result than on the old branch for the paper, for the same input files.

**Comment by WadimKoslow** [comment]
> 
> 
> > what do you mean by different initial values? Different for each run, or different compared to the results of our last paper?
> 
> calling read_population on current main gives a different result than on the old branch for the paper, for the same input files.

Hmm, I don't know why it should be different, I'll take a closer look at it.

**Comment by mknaranja** [comment]
@dabele There was also some kind of extrapolation for ICU and different age groups on the paper branch. Even if its not 100% reproducible then, I think this should be kept from the main branch.

**Review by WadimKoslow — COMMENTED** [review]
> 
> 
> @dabele There was also some kind of extrapolation for ICU and different age groups on the paper branch. Even if its not 100% reproducible then, I think this should be kept from the main branch.

ah, yes, that would explain it. Are only the ICU and Recovered values different?

**Comment by dabele** [comment]
yes, only ICU and recovered (and Susceptible of course) are different.

File: cpp/sims/paper_202011.cpp [hunk 2]
```diff
@@ -0,0 +1,877 @@
...
+    //- start and end date (may be incompatible with runmode::load)
+    //- seeds
+    //- log level
+    //- ...
+
+    epi::set_log_level(epi::LogLevel::warn);
+
+    RunMode mode;
+    std::string save_dir;
+    std::string data_dir;
```
[the remark this answers is not in the document]
Comment (dabele) [reply cpp/sims/paper_202011.cpp:837]: this is deliberate. It's almost never a good idea to use a fixed path inside a binary. some users may not want to use a different path (outside the repository) for the data, and a fixed path breaks easily.

**Review by WadimKoslow — APPROVED** [review]

**Pull request merged**